_source/_posts/2020-10-26-java-intellij-idea.adoc

```diff
@@ -0,0 +1,114 @@
+---
+layout: blog_post
+title: "Modern Java with IntelliJ IDEA"
+author: matt-raible
+by: advocate
+communities: [java]
+description: "Are you a Java developer that hasn't tried IntelliJ IDEA? Here's some reasons you might want to check it out."
+tags: [java, intellij, idea]
+tweets:
+- ""
+- ""
+- ""
+image:
+type: awareness
+---
+:toc: macro
+:page-liquid:
+:experimental:
+
+////
+Keywords: java intellij (5400)
+Title analyzer: https://user:APIH%40x0r@brentisarulebreaker.dev/
+- Modern Java with IntelliJ IDEA - 47
+- Modern Java Development with IntelliJ IDEA - 55
+- Why I Love Java + IntelliJ IDEA - 64
+- Why I Love IntelliJ IDEA for Java Development - 65
+////
+
+If you're a Java developer like me, you like to get shit done. I like a lot of things about IntelliJ IDEA, but I find the following ones make me most productive.
+
+toc::[]
+
+Disclaimer: as a Java Champion, I get a free license to IntelliJ. If I didn't my company would gladly pay for it.
+
+== Run Your Java Apps from IntelliJ IDEA
+
+Show how to run Spring Boot, Micronaut, and Quarkus from IntelliJ.
+
+Show how to set environment variables.
+
+Spring Dashboard for Spring Boot.
+
+New Project dialog:
+
+image::{% asset_path 'blog/java-intellij-idea/new-project.png' %}[alt=New Project Dialog,width=800,align=center]
+
+When I was a junior Java developer, I always wanted to use Ant or Maven to avoid vendor lock-in. In fact, I switched to IntelliJ in 2006 when I was migrating AppFuse from Ant to Maven. The Maven support for multiple modules was so much better than Eclipse's m2e, I switched.
```

Comment:
I worked at the company that was building the original m2e, we felt we had to use Eclipse, fortunately, one of my teammates told me to try IDEA.  I've been a much happier developer since then.